spec/jobs/petition_pdf_generation_worker_spec.rb

```diff
@@ -0,0 +1,38 @@
+require "rails_helper"
+
+RSpec.describe PetitionPdfGenerationWorker do
+  let(:petition_pdf_service) { instance_spy PetitionPdfService }
+  let(:detail_version_repository) { instance_spy PetitionPlugin::DetailVersionRepository }
+  let(:worker) { described_class.new petition_pdf_service: petition_pdf_service, repository: detail_version_repository }
+
+  describe "#perform" do
+    let!(:cycle) { create_cycle_with_phase phases: [{ plugin_type: :petition }] }
```

Comment:
Wouldn't be better as I said ealier just to mock the database here, by mocking your `repository` response?